### DISCUSSION TITLE
How valid new password

### DISCUSSION THREAD

**Discussion by caosdp-rs** [codeigniter4/shield#281] [Q&A] [answered]
How can I validate that the new password entered meets the requirements?

**Comment by caosdp-rs** [comment]
sample cde
        $user = auth()->user();
        $user->setPassword('123456');
        $user->saveEmailIdentity();

**Comment by caosdp-rs** [comment, marked as answer]
Is this the best implementation?

```php
$passwords = service('passwords');

if(!$passwords->check('123456',$user)->isOK()){
            print('Invalid password');
        };
```

**Reply by kenjis** [reply]
See https://github.com/codeigniter4/shield/blob/develop/docs/authentication.md#session-authenticator

**Comment by lonnieezell** [comment]
Maybe it’s not documented but you should be able to use the “strong_password” validation rule. I thinks that’s it. On mobile currently.
